omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/AllergyResource2_0.java

```diff
@@ -0,0 +1,219 @@
+			description.addProperty("auditInfo", findMethod("getAuditInfo"));
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getCreatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("patient");
+		description.addProperty("comments");
+
+		return description;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#newDelegate()
+	 */
+	@Override
+	public Allergy newDelegate() {
+		return new Allergy();
+	}
+
+	/**
+	 * Display string for allergy
+	 * 
+	 * @param allergy
+	 * @return String Allergen
+	 */
+	@PropertyGetter("display")
+	public String getDisplayString(Allergy allergy) {
+		if (allergy.getAllergen() == null)
+			return "";
+
+		return allergy.getAllergen().getCodedAllergen().getName().getName();
+	}
+
+	/**
+	 * Annotated setter for allergen
+	 *
+	 * @param allergy
+	 * @param  String Allergen
+	 */
+	@PropertySetter("allergen")
+	public static void setAllergen(Allergy allergy, Object value) {
+		Allergen allergen = new Allergen();
+		allergen.setCodedAllergen(Context.getConceptService().getConceptByUuid(
+				(String) value));
+		allergy.setAllergen(allergen);
+	}
+
+	/**
+	 * Annotated getter for allergen
+	 * 
+	 * @param allergy
+	 * @return String Allergen
+	 */
+	@PropertyGetter("allergen")
+	public static Allergen getAllergen(Allergy allergy) {
+		if (allergy.getAllergen() == null)
+			return null;
+		return allergy.getAllergen();
+	}
+
+	/**
+	 * Gets allergies for a given patient (paged according to context if
+	 * necessary) only if a patient parameter exists in the request set on the
+	 * {@link RequestContext}
+	 * 
+	 * @param context
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#doSearch(RequestContext)
+	 */
+	@Override
+	protected PageableResult doSearch(RequestContext context) {
+		String patientUuid = context.getRequest().getParameter("patient");
+		if (patientUuid != null) {
+			Patient patient = ((PatientResource1_8) Context.getService(
+					RestService.class).getResourceBySupportedClass(
+					Patient.class)).getByUniqueId(patientUuid);
+			if (patient == null)
+				return new EmptySearchResult();
+
+			return new NeedsPaging<Allergy>(Context.getPatientService()
+					.getAllergies(patient), context);
+		}
+
+		// currently this is not supported since the superclass throws an
+		// exception
+		return super.doSearch(context);
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getByUniqueId(java.lang.String)
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public Allergy getByUniqueId(String uniqueId) {
+		// TODO cannot be implemented because Allergies doesn't have a Patient
```

Comment:
You should call getPatientService.getAllergyByUuid here. I see that there's no such method in openmrs-core, but it must be simply added there.